### PR TITLE
[CASSH Server] v1.8.0: Python3 support

### PR DESCRIPTION
**Pull request by nbeguier**
CASSH Server
-----

1.8.0
-----

2019/05/27

### New Features
  - Python 3.6 support :
    - remove  __future__
    - use urllib.parse instead of urllib for unquote_plus
    - web.data output in encoded in UTF-8
    - Used .keys() for dict
    - write temporary files in unicode
    - check_output in returning an unicode output

### Changes
  - Python 2.x deprecated
  - Used web.py version 0.40-dev1 instead of 0.39

1.7.3
-----

2019/05/27

### New Features
  - Add debug parameters in configuration

### Changes
  - Use ldap version 3.2.0 instead of 2.5.2 (open => initialize)
  - Edit Dockerfile and requirements.txt
  - Tools: Rename cluster_updatekrl into cluster_update_krl

1.7.2
-----

2019/05/24

### Changes
  - split tools functions in another library

1.7.1
-----

2019/05/23

### Changes
  - always return a Content-Type
  - Block bad realnames (XSS stored)
  - Doesn-t return a blocked username (XSS reflected)

### Bug Fixes
  - Fix some missing http code
  - Fix according tests

### CODE REVIEW

**Comment by nbeguier** [comment]
@xakraz I've found the problem with the tests, I'm fixing it

File: src/server/Dockerfile [hunk 1]
```diff
@@ -1,19 +1,19 @@
-FROM debian:latest
+FROM python:3.6
```
Comment (xakraz) [inline src/server/Dockerfile:1]: :+1:

File: src/server/requirements.txt [hunk 2]
```diff
@@ -1,4 +1,5 @@
-python-ldap
-psycopg2
-urllib
-web.py
\ No newline at end of file
+configparser==3.7.4
```
Comment (xakraz) [inline src/server/requirements.txt:1]: pinning :tada:

File: src/server/tools/__init__.py [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+        string = unquote_plus(string)
+        if '+' in string and '%20' in string:
+            # Old custom quote
+            string = string.replace('%20', ' ')
+    return string
+
+
+class Tools(object):
+    """
+    Class tools
```
Comment (xakraz) [inline src/server/tools/__init__.py:110]: Tools seems a bit too blurry to me... It does a lot of things and the description is not clear enough.

### 1st 

You did not include these 2 functions which are still in the main python file ?
- `ldap_authentification`
- `data2map`


### 2nd, about the scopes

What do you think about:
* A `Server` Class that has the CASSH specific methods
  - `list_keys`
  - `sign_keys`
* A `ClusterController` Class with:
  - `status` (`cluster_alived`)
  - `get_krl` (`cluster_last_krl`)
  - `update_krl` (`cluster_update_krl`)

Maybe `Server` would merge into `ClusterController` at some point and just be renamed as `Controller` with a option / param to support cluster mode.

And finally the `Tools` class could be removed and these methods would go back as functions :
  - `pg_connection`
  - `get`
  - `post`
  - `sql_to_json`

and `tools`  might me renamed as `cassh_utils` for the functions needed outside of him ?

**Review by xakraz — APPROVED** [review]
The main features advertised in the Changelog LGTM.

I am just a bit reserved about the new `Tools` class ...

File: src/server/tools/__init__.py (hunk 3 above)
Comment (nbeguier) [reply src/server/tools/__init__.py:110]: I's a good idea. I'm creating an issue to clean the code in the future, with these insight

Comment (nbeguier) [reply src/server/tools/__init__.py:110]: https://github.com/nbeguier/cassh/issues/71

**Pull request merged**